tests/core/models/test_edition.py

```diff
@@ -325,6 +328,67 @@ def test_calculate_presentation_author(self, db: DatabaseTransactionFixture):
         assert "Kelly Accumulator, Bob A. Bitshifter" == wr.author
         assert "Accumulator, Kelly ; Bitshifter, Bob" == wr.sort_author
 
+    def test_calculate_presentation_very_long_author(
+        self, db: DatabaseTransactionFixture
+    ):
+        authors = []
+
+        # author names should be unique and not similar to ensure that the
+        # test mirrors the types of long author lists we'd expect in real data.
+        def generate_random_random_author():
+            return "".join(
+                random.choices(
+                    string.ascii_uppercase + string.ascii_lowercase + string.digits,
+                    k=25,
+                )
+            )
+
+        for i in range(0, 500):
+            author, ignore = db.contributor(sort_name=generate_random_random_author())
+            authors.append(author.sort_name)
+
+        untruncated_authors = ", ".join([x for x in sorted(authors)])
+        wr = db.edition(authors=authors)
+        wr.calculate_presentation()
+        db.session.commit()
+
+        def do_check(original_str: str, truncated_str: str):
+            assert (
+                len(truncated_str)
+                == Edition.SAFE_AUTHOR_FIELD_LENGTH_TO_AVOID_PG_INDEX_ERROR
+            )
+            assert truncated_str.endswith("...")
+            assert not original_str.endswith("...")
+            assert (
+                len(original_str)
+                > Edition.SAFE_AUTHOR_FIELD_LENGTH_TO_AVOID_PG_INDEX_ERROR
+            )
+
+        do_check(untruncated_authors, wr.author)
+        do_check(untruncated_authors, wr.sort_author)
```

Comment:
On l. 368, I wouldn't expect this test to pass because it's comparing authors to sort authors: `do_check(untruncated_authors, wr.sort_author)`. But maybe that's got something to do with my comment on l. 347, above.

Reply:
It's not that.  It passes because the author name is set to the sort_author value if it is not specified when creating the contributor (https://github.com/ThePalaceProject/circulation/blob/main/tests/fixtures/database.py#L431).

Reply:
That can be confusing when trying to understand the test.

Reply:
You might consider using a separate variable (or a shared variable that is more clearly general between sorted and non-sorted) or add a comment to clarify what's happening.